### ISSUE TITLE
Remove nightly off-site backup of Asset Manager assets from NFS to S3

### ISSUE THREAD

**Issue by floehopper** [alphagov/asset-manager#295]
Once we're happy that [cross-site replication from the `govuk-assets-production` S3 bucket to the `govuk-assets-backup-production` S3 bucket](https://github.com/alphagov/asset-manager/issues/146) is working OK, we should be able to remove [the `asset-manager-s3` job](https://github.com/alphagov/govuk-puppet/blob/55282a2b423d69af26d17d92f91887a035efa057/hieradata/production.yaml#L21-L30) which does a nightly backup of Mainstream assets from NFS to the `govuk-offsite-backups-production` S3 bucket under `/asset-manager/*`.

We should also remove any other related code from `govuk-puppet`.

Note that [the `assets-whitehall-s3` job](https://github.com/alphagov/govuk-puppet/blob/55282a2b423d69af26d17d92f91887a035efa057/hieradata/production.yaml#L8-L20) should stay in place until all existing Whitehall assets have been migrated to Asset Manager.

**Comment by floehopper** [comment]
I've opened https://github.com/alphagov/govuk-puppet/pull/6768 as a first step to removing the nightly backup job.

**Comment by floehopper** [comment]
@chrisroos and I looked through the code in the PR and thought it might be worth seeing whether the cron job for the off-site backup for Asset Manager assets is currently present (as we expect) in the `root` user's crontab on the `asset-slave-1` machine in production. We could then check that the change in this PR removes that cron job.

**Comment by floehopper** [comment]
The cron job does indeed appear in the crontab for the `root` user on the `asset-slave-1` machine in production:

```
# HEADER: This file was autogenerated at 2017-09-08 11:08:22 +0000 by puppet.
# HEADER: While it can still be managed manually, it is definitely not recommended.
# HEADER: Note particularly that the comments starting with 'Puppet Name' should
# HEADER: not be deleted, as doing so could cause duplicate cron jobs.
# Puppet Name: tmpreaper-bulk-upload-zip-file-tmp
5 5 * * * /usr/sbin/tmpreaper -am 24h /mnt/uploads/whitehall/bulk-upload-zip-file-tmp/
# Puppet Name: tmpreaper-attachment-cache
25 5 * * * /usr/sbin/tmpreaper -T300 --mtime 14d /mnt/uploads/whitehall/attachment-cache/
# Puppet Name: tmpreaper-carrierwave-tmp
15 5 * * * /usr/sbin/tmpreaper -T300 --mtime 7d /mnt/uploads/whitehall/carrierwave-tmp/
# Puppet Name: puppet
28,58 * * * * /usr/local/bin/govuk_puppet
# Puppet Name: assets-whitehall-s3
20 4 * * * /var/spool/duplicity/assets-whitehall-s3.sh
# Puppet Name: asset-manager-s3
13 4 * * * /var/spool/duplicity/asset-manager-s3.sh
```

It's the `asset-manager-s3` job which we're interested in - this should disappear once these changes are merged and deployed to production.

**Comment by floehopper** [comment]
Note that we will need to follow-up https://github.com/alphagov/govuk-puppet/pull/6768 with another PR to remove the configuration for [the `assets-whitehall-s3` job](https://github.com/alphagov/govuk-puppet/blob/55282a2b423d69af26d17d92f91887a035efa057/hieradata/production.yaml#L8-L20).

**Comment by floehopper** [comment]
We should also consider updating relevant documentation pages on https://docs.publishing.service.gov.uk.

**Comment by chrisroos** [comment]
We spoke to @stephenharker about this today and he's keen for us to retain these incremental archive backups from S3. The backups should be stored somewhere other than Amazon although the actual destination is to be decided.

**Comment by floehopper** [comment]
I'm no longer actively working on this, because we're waiting until we hear what the plan is with the backups now we're switched to S3.

**Comment by floehopper** [comment]
I've now captured the idea of replacing the Duplicity backups in a [new issue](https://github.com/alphagov/asset-manager/issues/347). We'll still almost certainly want to remove the existing Duplicity jobs, so I'm going to leave this issue in place.

**Comment by floehopper** [comment]
As I mentioned in [this issue](https://github.com/alphagov/asset-manager/issues/347#issuecomment-353387417), we're only planning to switch off these backups shortly before we start deleting files from the Asset Manager part of the NFS mount. This is so we end up with a useful backup of the files we're going to delete.

**Comment by floehopper** [comment]
https://github.com/alphagov/govuk-puppet/pull/6768 has been merged.

**Comment by floehopper** [comment]
These changes have just been deployed to staging.

**Comment by floehopper** [comment]
These changes have just been deployed to production.

**Comment by floehopper** [comment]
This has now been applied in production and I can see that the `crontab` for the `root` user on `asset-slave-1` in production that the `asset-manager-s3` cron job has been removed, but the `assets-whitehall-s3` cron job has been left untouched as expected.

```
# HEADER: This file was autogenerated at 2018-01-08 14:28:45 +0000 by puppet.
# HEADER: While it can still be managed manually, it is definitely not recommended.
# HEADER: Note particularly that the comments starting with ‘Puppet Name’ should
# HEADER: not be deleted, as doing so could cause duplicate cron jobs.
# Puppet Name: tmpreaper-bulk-upload-zip-file-tmp
5 5 * * * /usr/sbin/tmpreaper -am 24h /mnt/uploads/whitehall/bulk-upload-zip-file-tmp/
# Puppet Name: tmpreaper-attachment-cache
25 5 * * * /usr/sbin/tmpreaper -T300 --mtime 14d /mnt/uploads/whitehall/attachment-cache/
# Puppet Name: tmpreaper-carrierwave-tmp
15 5 * * * /usr/sbin/tmpreaper -T300 --mtime 7d /mnt/uploads/whitehall/carrierwave-tmp/
# Puppet Name: puppet
28,58 * * * * /usr/local/bin/govuk_puppet
# Puppet Name: assets-whitehall-s3
20 4 * * * /var/spool/duplicity/assets-whitehall-s3.sh
```

Closing.
